htdocs/ticket/card.php

```diff
@@ -381,20 +381,18 @@
 
 		if (!$error) {    // Update list of contacts
 			// Si déjà un user assigné on le supprime des contacts
-			if ($useroriginassign > 0) {
+			if ($useroriginassign > 0 && empty($conf->global->TICKET_DONOT_DELETE_UNASSIGNED_CONTACT)) {
 				$internal_contacts = $object->listeContact(-1, 'internal');
 
 				foreach ($internal_contacts as $key => $contact) {
 					if ($contact['code'] == "SUPPORTTEC" && $contact['id'] == $useroriginassign) {
-					}
-					{
 						//print "user à effacer : ".$useroriginassign;
 						$object->delete_contact($contact['rowid']);
 					}
 				}
 			}
 
-			if ($usertoassign > 0) {
+			if ($usertoassign > 0 && $usertoassign != $useroriginassign) {
```

Comment:
At line 384, we force the deletion of link contact for $useroriginassign, so we can make the add_contact at no risk to insert it twice. But if we add test  $usertoassign != $useroriginassign  and we assign to them, we first delete and then we do not recreate it loosing the contact, right ?